lib/ansible/plugins/action/__init__.py

```diff
@@ -1033,3 +1024,33 @@ def _find_needle(self, dirname, needle):
 
         # if missing it will return a file not found exception
         return self._loader.path_dwim_relative_stack(path_stack, dirname, needle)
+
+    def _get_admin_users(self):
+        '''
+        Returns a list of admin users that are configured for the current shell
+        plugin
+        '''
+        # fallback for old custom plugins w/o get_option
+        try:
+            admin_users = self._connection._shell.get_option('admin_users')
+        except AnsibleError:
+            admin_users = ['root']
+        return admin_users
+
+    def _deescalated_user(self):
+        '''
+        The user is not the same as the connection user and is not part of the
+        shell configured admin users
+        '''
+        # if we don't use become then we know the user is deescalated
+        if not self._play_context.become:
+            return False
+
+        # if we use become and the user is not an admin (or same user) then
+        # we need to deescalate
+        admin_users = self._get_admin_users()
+        try:
+            remote_user = self._connection.get_option('remote_user')
+        except AnsibleError:
+            remote_user = self._play_context.remote_user
+        return bool(self._play_context.become_user not in admin_users + [remote_user])
```

Comment:
In general, (except for __init__) our structure is for callers to be towards the bottom of files and callees to be towards the top.  So these should be somewhere above _make_tmp_path() in the class.

Reply:
No worries, is this just an Ansible thing or common in Python?

Reply:
I've seen it in Python more frequently than putting callers at the top and callees below.  I know I got this style because  I came in via Pascal, C, and C++ which all use it (At least, at the time when I learned C++).  I'm not sure if that has any bearing on why other Python programmers/projects use it but it does seem to be the most popular option.